homework/week4/2025320058-twuel0921/No9663.java

```diff
@@ -0,0 +1,36 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+//N-Queen 풀이 코드
+public class No9663 {
+    private static int n;
+    public static void main(String args[]) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        n = Integer.parseInt(br.readLine());
+        int[] col = new int[n+1];
+        System.out.println(findNQueen(0, col));
+    }
+    private static int findNQueen(int i, int[] col) {
+        int cnt_nQueen = 0;
+        if(isPromising(i, col)) {
```

Comment:
이게 아닐때 return하면, 코드가 조금 더 깔끔해질 듯 합니다.

Reply:
이 함수에서 if가 시작부터 끝까지 있는 것으로 파악했습니다. 아래처럼 고치면 어떨까 싶습니다.
```suggestion
        if(!isPromising(i, col)) {
            return 0;
        }
        // if문 안에 있던 코드들
```

Reply:
아, 제가 이해를 잘못했었군요. 고쳐서 커밋하도록 하겠습니다!